tests/validation/tests/v3_api/resource/terraform/rke2/master/define_node_role.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+# This script defines which role this node will be and writes that to a file
+# that is readable by rke2
+
+
+if [ $# != 8 ]; then
+  echo "Please see $0 for the variables to pass to the script."
+  exit 1
+else
+  let node_index=$1+1
+  role_order=$2
+  all_role_nodes=$3
+  etcd_only_nodes=$4
+  etcd_cp_nodes=$5
+  etcd_worker_nodes=$6
+  cp_only_nodes=$7
+  cp_worker_nodes=$8
+fi
+
+# Set the desired role into an array based on the index
+order_array=($(echo "$role_order" | tr ',' '\n'))
+role_array=()
+for order_num in "${order_array[@]}"
+do
+  if [[ "$order_num" = "1" ]]
+  then
+    until [ $all_role_nodes -le "0" ]
+    do
+       role_array+=("all-roles")
+       let "all_role_nodes-=1"
+    done
+  elif [[ "$order_num" = "2" ]]
+  then
+    until [ $etcd_only_nodes -le "0" ]
+    do
+       role_array+=("etcd-only")
+       let "etcd_only_nodes-=1"
+    done
+  elif [[ "$order_num" = "3" ]]
+  then
+    until [ $etcd_cp_nodes -le "0" ]
+    do
+       role_array+=("etcd-cp")
+       let "etcd_cp_nodes-=1"
+    done
+  elif [[ "$order_num" = "4" ]]
+  then
+    until [ $etcd_worker_nodes -le "0" ]
+    do
+       role_array+=("etcd-worker")
+       let "etcd_worker_nodes-=1"
+    done
+  elif [[ "$order_num" = "5" ]]
+  then
+    until [ $cp_only_nodes -le "0" ]
+    do
+       role_array+=("cp-only")
+       let "cp_only_nodes-=1"
+    done
+  elif [[ "$order_num" = "6" ]]
+  then
+    until [ $cp_worker_nodes -le "0" ]
+    do
+       role_array+=("cp-worker")
+       let "cp_worker_nodes-=1"
+    done
+  fi
+done
+
```

Comment:
Nit-picky: Can we add a check that if any of those values (etcd_only_nodes, etcd_cp_nodes, etc) are > 0, their corresponding number should exist in RKE2_ROLE_ORDER?